packages/timeline-state-resolver-types/src/tricaster.ts

```diff
@@ -0,0 +1,307 @@
+import { Mapping } from './mapping'
+import { DeviceType, TimelineDatastoreReferencesContent, TSRTimelineObjBase } from '.'
+
+export type TriCasterMixEffectName = 'main' | `v${number}`
+export type TriCasterKeyerName = `dsk${number}`
```

Comment:
But bigint would still allow negative ints, 0 and values higher than the number of the available dsks. I don't think this is that much of a problem if these values are validated in the TimelineStateConverter against resources available in the video switcher that we're connected to. Wasn't bigint made to solve a totally different problem?

Reply:
Both arguments are true. BigInt was made to solve big number representation. But for the type, it will make it more concrete in the sense that we will not allow floats formatted as strings. It is just to concretize the input domain.
As long as it is checked, it should be okay.